src/core/TawktoGenerator.php

```diff
@@ -54,24 +54,45 @@ public function getWidget() {
     $display_opts = $options;
     // Default value.
     $enable_visitor_recognition = TRUE;
+    $secure_mode_enabled = FALSE;
+    $js_api_key = NULL;
     if (!is_null($display_opts)) {
       $display_opts = json_decode($display_opts);
 
       if (!is_null($display_opts->enable_visitor_recognition)) {
         $enable_visitor_recognition = $display_opts->enable_visitor_recognition;
       }
+
+      if (!is_null($display_opts->secure_mode_enabled)) {
+        $secure_mode_enabled = $display_opts->secure_mode_enabled;
+      }
+
+      if (!is_null($display_opts->js_api_key)) {
+        $js_api_key = $display_opts->js_api_key;
+      }
     }
 
     if ($enable_visitor_recognition) {
       $user = User::load(\Drupal::currentUser()->id());
       if ($user) {
         $username = $user->get('name')->value;
         $usermail = $user->get('mail')->value;
+        $hash = NULL;
+
+        $keyModule = $this->getKeyModule();
+        $keys = $keyModule['keys'];
```

Comment:
Instead of getting all keys which could be costly for installations with higher key counts, simply get key directly

https://www.drupal.org/docs/contributed-modules/key/developer-guide#s-get-a-key-value